src/middlewared/middlewared/plugins/pool.py

```diff
@@ -657,17 +679,25 @@ def pool_extend(self, pool):
         if not os.path.isdir(cachefile_dir):
             os.makedirs(cachefile_dir)
 
-        job.set_progress(90, 'Creating ZFS Pool')
-        z_pool = await self.middleware.call('zfs.pool.create', {
-            'name': data['name'],
-            'vdevs': vdevs,
-            'options': options,
-            'fsoptions': fsoptions,
-        })
-
-        job.set_progress(95, 'Setting pool options')
-        pool_id = None
+        pool_id = z_pool = None
         try:
+            await self.middleware.call(
+                'pool.dataset.insert_or_update', {
+                    'name': data['name'], 'encryption_key': encryption_dict.get('key'),
+                    'key_format': encryption_dict.get('keyformat')
+                }
+            )
```

Comment:
So we do delete it in case an exception occurs below using the dataset name. I saved it in the database before even the pool is created for a hard edge case where for any reason the process is disrupted and we lose keys to the encrypted root dataset. But it doesn't make much sense still to do this first. I'll correct the order